algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py

```diff
@@ -0,0 +1,229 @@
+    pass
+
+  @property
+  def param_shapes(self):
+    """The shapes of the parameters in the workload model."""
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    return self._param_shapes
+
+  @property
+  def model_params_types(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    if self._param_types is None:
+      self._param_types = param_utils.jax_param_types(self._param_shapes)
+    return self._param_types
+
+  def output_activation_fn(self,
+                           logits_batch: spec.Tensor,
+                           loss_type: spec.LossType) -> spec.Tensor:
+    """Return the final activations of the model."""
+    pass
+
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,  # Dense (not one-hot) labels.
+      logits_batch: spec.Tensor,
+      mask_batch: Optional[spec.Tensor] = None) -> spec.Tensor:
+    per_example_losses = metrics.per_example_sigmoid_binary_cross_entropy(
+        logits=logits_batch, targets=label_batch)
+    if mask_batch is not None:
+      weighted_losses = per_example_losses * mask_batch
+      normalization = mask_batch.sum()
+    else:
+      weighted_losses = per_example_losses
+    normalization = label_batch.shape[0]
+    return jnp.sum(weighted_losses, axis=-1) / normalization
+
+  def init_model_fn(self, rng: spec.RandomState) -> spec.ModelInitState:
+    rng, init_rng = jax.random.split(rng)
+    init_fake_batch_size = 2
+    input_size = _NUM_DENSE_FEATURES + len(_VOCAB_SIZES)
+    input_shape = (init_fake_batch_size, input_size)
+    target_shape = (init_fake_batch_size, input_size)
+
+    initial_variables = jax.jit(self._flax_module.init)(
+        init_rng,
+        jnp.ones(input_shape, jnp.float32),
+        jnp.ones(target_shape, jnp.float32))
+
+    initial_params = initial_variables['params']
+    self._param_shapes = jax.tree_map(lambda x: spec.ShapeTuple(x.shape),
+                                      initial_params)
+    return jax_utils.replicate(initial_params), None
+
+  def model_fn(
+      self,
+      params: spec.ParameterContainer,
+      augmented_and_preprocessed_input_batch: Dict[str, spec.Tensor],
+      model_state: spec.ModelAuxiliaryState,
+      mode: spec.ForwardPassMode,
+      rng: spec.RandomState,
+      update_batch_norm: bool) -> Tuple[spec.Tensor, spec.ModelAuxiliaryState]:
+    del model_state
+    del mode
+    del rng
+    del update_batch_norm
+    inputs = augmented_and_preprocessed_input_batch['inputs']
+    targets = augmented_and_preprocessed_input_batch['targets']
+    logits_batch = self._flax_module.apply({'params': params}, inputs, targets)
+    return logits_batch, None
+
+  @property
+  def step_hint(self):
+    return 64_000  # DO NOT SUBMIT
```

Comment:
Probably also a `TODO`

Reply:
Done (I had a commit that changed these that I meant to push before emailing you, but forgot to hit "enter" on the git push command :upside_down_face: )